src/reducers/comments.tsx

```diff
@@ -156,6 +170,73 @@ export const selectVersionHasComments = ({
   return Object.keys(comments.byId).length > 0;
 };
 
+export const fetchAndLoadComments = ({
+  _getComments = getComments,
+  addonId,
+  versionId,
+}: {
+  _getComments?: typeof getComments;
+  addonId: number;
+  versionId: number;
+}): ThunkActionCreator => {
+  return async (dispatch, getState) => {
+    const { api: apiState, comments } = getState();
+
+    if (comments.isLoading) {
+      return;
+    }
+
+    dispatch(actions.beginLoadVersionComments({ versionId }));
+
+    // TODO: fetch all pages to get all comments.
+    // https://github.com/mozilla/addons-code-manager/issues/1093
+    const response = await _getComments({ addonId, apiState, versionId });
+
+    if (isErrorResponse(response)) {
+      dispatch(actions.abortLoadVersionComments({ versionId }));
+      dispatch(errorsActions.addError({ error: response.error }));
+      return;
+    }
+
+    const map: { [key: string]: ExternalComment[] } = {};
+
+    for (const comment of response.results) {
+      let key = '';
+      if (comment.filename) {
+        key = comment.filename;
+      }
+      if (comment.lineno) {
+        key = `${key}:${comment.lineno}`;
+      }
+      if (!map[key]) {
+        map[key] = [];
+      }
+      map[key].push(comment);
+    }
+
+    for (const key of Object.keys(map)) {
+      let line = null;
+      let fileName = null;
+      if (key) {
+        const parts = key.split(':');
+        fileName = parts[0];
+        if (parts[1]) {
+          line = parseInt(parts[1], 10);
+        }
+      }
```

Comment:
I don't understand this part to be honest, probably because I am not familiar with the API response. Maybe add a comment to explain why we want to build a map before calling `setComments()`. Is it because comments are not sorted in the response and we want to batch set comments by filename and line?

Reply:
One more thing: instead of using the key to retrieve `fileName` and `line`, how about changing `map` to:

```js
const map: { [key: string]: {
  fileName: string,
  line: number | null,
  comments: ExternalComment[],
 } } = {};
```

Reply:
I'll add a comment. I wanted to minimize the number of dispatches since every Redux dispatch hits all the reducers. The API returns a list of comment objects and what this mapping code does is sort them into version comments, file comments, and line comments to reduce the number of dispatches.

Reply:
> One more thing: instead of using the key to retrieve `fileName` and `line`, how about changing `map` to:
> 
> ```js
> const map: { [key: string]: {
>   fileName: string,
>   line: number | null,
>   comments: ExternalComment[],
>  } } = {};
> ```

Thanks, this is a lot easier to deal with.